src/messageschannel.cpp

```diff
@@ -43,10 +42,11 @@ MatrixMessagesChannel::MatrixMessagesChannel(MatrixConnection *connection, Tp::B
     QStringList supportedContentTypes = QStringList() << QStringLiteral("text/plain");
     const QList<uint> messageTypes = {
         Tp::ChannelTextMessageTypeNormal,
+        Tp::ChannelTextMessageTypeDeliveryReport,
     };
 
-    uint messagePartSupportFlags = 0; // Tp::MessageSendingFlagReportDelivery | Tp::MessageSendingFlagReportRead;
-    uint deliveryReportingSupport = 0; // Tp::DeliveryReportingSupportFlagReceiveSuccesses | Tp::DeliveryReportingSupportFlagReceiveRead;
+    uint messagePartSupportFlags = Tp::MessageSendingFlagReportDelivery | Tp::MessageSendingFlagReportRead;
+    uint deliveryReportingSupport = Tp::DeliveryReportingSupportFlagReceiveSuccesses | Tp::DeliveryReportingSupportFlagReceiveRead;
```

Comment:
I think we need to add ReceiveFailures to let a client expect the failures reports.
It seems that in Qt style it is going to look like this:
```
    uint deliveryReportingSupport = Tp::DeliveryReportingSupportFlagReceiveFailures
            | Tp::DeliveryReportingSupportFlagReceiveSuccesses
            | Tp::DeliveryReportingSupportFlagReceiveRead;
```
Frankly speaking I have no idea if a client actually need it, but let's follow the spec.